### PR TITLE
#690 Guardian Verify Backups

### LINKED ISSUE #690: ✨ Guardian Verify Backups in GUI (#6d)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

As a guardian when backups have been shared from the admin the system should verify those backups and send the verifications back out to the admins via the DB

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lprichar**
### Issue

Fixes #690 

### Description
Once the admin receives and shares backups the guardians will now verify the backups and send the verifications back to the admin.  This PR additionally includes a variety of linting fixes after adding mypy linting back to the gui, so there is more churn than usual.

### Testing
1. Create a key ceremony in the GUI for an admin
2. Have guardians join the key ceremony
3. The UI should update to get to a state like this:

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,7 +97,7 @@ blackcheck:
 	poetry run black --check .
 
 mypy:
-	poetry run mypy src/electionguard src/electionguard_tools src/electionguard_cli stubs
+	poetry run mypy src/electionguard src/electionguard_tools src/electionguard_cli src/electionguard_gui stubs
```
Comment (lprichar) [inline Makefile:100]: Sorry this ^ generated a lot of churn.  I should have added it sooner.

File: src/electionguard/guardian.py [hunk 2]
```diff
@@ -363,8 +363,10 @@ def verify_election_partial_key_backup(
         """
         backup = self._guardian_election_partial_key_backups.get(guardian_id)
         public_key = self._guardian_election_public_keys.get(guardian_id)
-        if backup is None or public_key is None:
-            return None
+        if backup is None:
+            raise ValueError(f"No backup exists for {guardian_id}")
+        if public_key is None:
+            raise ValueError(f"No public key exists for {guardian_id}")
```
Comment (lprichar) [inline src/electionguard/guardian.py:369]: Once upon a time the electionguard repo avoided throwing exceptions to ease of transcribing to languages that don't support exceptions.  I see no reason to continue that pattern, but I will revert if anyone disagrees.

File: src/electionguard_gui/components/create_key_ceremony_component.py [hunk 3]
```diff
@@ -44,8 +47,8 @@ def create_key_ceremony(
         )
         if existing_key_ceremonies:
             self.log.debug(f"record '{key_ceremony_name}' already exists")
-            result: dict[str, Any] = eel_fail("Key ceremony name already exists")
-            return result
+            fail_result: dict[str, Any] = eel_fail("Key ceremony name already exists")
```
Comment (lprichar) [inline src/electionguard_gui/components/create_key_ceremony_component.py:50]: `mypy` is occasionally unable to determine types when you return a function call directly.  This is a bit annoying, but this seems to fix it.

File: src/electionguard_gui/services/key_ceremony_stages/key_ceremony_s5_verify_backup_service.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from pymongo.database import Database
+from electionguard.key_ceremony import ElectionPartialKeyVerification
+from electionguard_gui.models.key_ceremony_dto import KeyCeremonyDto
+from electionguard_gui.models.key_ceremony_states import KeyCeremonyStates
+from electionguard_gui.services.key_ceremony_stages.key_ceremony_stage_base import (
+    KeyCeremonyStageBase,
+)
+
+
+class KeyCeremonyS5VerifyBackupService(KeyCeremonyStageBase):
```
Comment (lprichar) [inline src/electionguard_gui/services/key_ceremony_stages/key_ceremony_s5_verify_backup_service.py:11]: This is where the real core logic for this PR resides.

File: src/electionguard_gui/services/authorization_service.py [hunk 5]
```diff
@@ -16,6 +16,11 @@ def expose(self) -> None:
         eel.expose(self.set_user_id)
         eel.expose(self.is_admin)
 
+    def get_required_user_id(self) -> str:
+        if self.user_id is None:
+            raise Exception("Use must be logged in")
```
Comment (SteveMaier-IRT) [inline src/electionguard_gui/services/authorization_service.py:21]: Small typo.... User not Use

File: src/electionguard/guardian.py (hunk 2 above)
Comment (SteveMaier-IRT) [reply src/electionguard/guardian.py:369]: Agreed.  Exceptions will be put back until it needs to be discussed.

**Review by SteveMaier-IRT — APPROVED** [review]
looks good

**Pull request merged**